Add Valgrind plugin

jamestiotio opened the pull request:
This commit adds the Valgrind plugin to Ceedling. The plugin implementation closely follows the Gcov plugin.

Fixes #277.

mvandervoord commented:
It would be better at this point if this were targeting the prerelease branch that is soon to be released.

jamestiotio commented:
> It would be better at this point if this were targeting the prerelease branch that is soon to be released.

@mvandervoord Sure, I have changed the base branch of the pull request to [test/ceedling_0_32_rc](/ThrowTheSwitch/Ceedling/tree/test/ceedling_0_32_rc).

I have also added a simple README file for the Valgrind plugin in the latest commit.

PasVegan commented:
Why is this still not merged ?

mvandervoord commented:
@PasVegan Because it hasn't been reviewed yet.

Have you tried the plugin? How was your experience?

PasVegan commented:
Yes I tried the plugin and it works very nicely, the only thing that I have to say is that even if valgrind reports errors the exit code is still 0 at the end. I think that need to be changed for proper CI use.

jamestiotio commented:
Hi @PasVegan, you can add the following flags to let Valgrind exit on the first error and return a different exit code:

```
--exit-on-first-error=yes
--error-exitcode=<number>
```

In a CI context, it makes sense to add said flags. If you feel that those flags should be the default options, I can add them (with the default error exit code to be 1).

PasVegan commented:
Hi, yes I think those flags should be on by default with the default exit code on error being 1 👍🏻

jamestiotio commented:
> Hi, yes I think those flags should be on by default with the default exit code on error being 1 👍🏻

Done! Added the 2 aforementioned flags in the latest commit (68fe5c33a233e52b371de18652aa9f64adfd32b8).

jamestiotio commented:
Hi there, @mvandervoord. It looks like these changes were not included in the master branch during the merge of the v1.0.0 release candidate. Could I kindly check if there is a reason for this? Thank you.
